python/grass/tools/session_tools.py

```diff
@@ -17,10 +17,82 @@
 import os
 
 import grass.script as gs
+from grass.exceptions import CalledModuleError
 
 from .support import ParameterConverter, ToolFunctionResolver, ToolResult
 
 
+class ToolError(CalledModuleError):
+    """Raised when a tool run ends with error (typically a non-zero return code)
+
+    Inherits from *subprocess.CalledProcessError* to make it easy to transition from
+    or combine with code which is using the *subprocess* package.
+    Inherits from CalledModuleError to make it easy to transition code with except
+    statements around *grass.script.run_command*-style tool calls, but new code should
+    not rely on that.
+    """
+
+    def __init__(self, tool, cmd, returncode, errors=None):
+        """Create an exception with a full error message based on the parameters.
+
+        Best results are provided when errors is a single line string, aiming at a
+        short and clear message. In case it is `None`, additional text is providing
+        to help the user find the error message, assuming that there is one somewhere.
+        If it is an empty string, it assumes that none was produced and reports
+        that in the resulting message. For multiline error messages,
```

Comment:
Just to be sure, "it" refers to the "errors"'s argument value? I reread like 5 times to understand 


And the lowercase none means none like in English, not Python `None`?